core/src/main/scala/akka/kafka/scaladsl/Consumer.scala

```diff
@@ -171,5 +172,9 @@ object Consumer {
     }
   }
 
+  def committablePartitionedSource[K, V](settings: ConsumerSettings[K, V]):
+    Source[(TopicPartition, Source[CommittableMessage[K, V], Control]), Control] = {
```

Comment:
I think SubFlow is inherently hard to extend, so that is a no-go. You can achieve the same with groupBy though and having a (Partition, Msg) pair and have your source-of-sources as an addition instead of the baseline API. I am fine with pure source-of-sources as well, but then Sources must have clear lifecycle semantics when they happen to be dropped somewhere and no-one consumes them.